python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -263,8 +264,10 @@ def __call__(self, fn: Callable) -> AssetsDefinition:
         with warnings.catch_warnings():
             warnings.simplefilter("ignore", category=ExperimentalWarning)
 
-            required_resource_keys = set(self.required_resource_keys).union(
-                set(self.resource_defs.keys())
+            required_resource_keys = (
+                set(self.required_resource_keys)
+                .union(set(self.resource_defs.keys()))
+                .union({arg.name for arg in get_resource_args(fn)})
```

Comment:
I think we should strongly consider disallowing specifying `required_resource_keys`/`resource_defs` and the argument-based approach for now, similar to how we do not allow `config` argument and `config_schema`. We can always loosen this later, but I would prefer to be strict for now.